File: replace_translations.js
const fs = require('fs');
const path = require('path');

const mappings = [
  // CEO
  ["{{ langService.currentLang() === 'en' ? 'Executive Overview' : 'نظرة عامة تنفيذية' }}", "{{ 'CEO.DASHBOARD_TITLE' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Real-time performance metrics' : 'مقاييس الأداء في الوقت الفعلي' }}", "{{ 'CEO.DASHBOARD_SUB' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Export Report' : 'تصدير التقرير' }}", "{{ 'CEO.EXPORT_REPORT' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? \"Today's Revenue\" : 'إيرادات اليوم' }}", "{{ 'CEO.TODAYS_REVENUE' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'vs yesterday' : 'مقارنة بالأمس' }}", "{{ 'CEO.VS_YESTERDAY' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Active Patients' : 'المرضى النشطين' }}", "{{ 'CEO.ACTIVE_PATIENTS' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'new today' : 'جديد اليوم' }}", "{{ 'CEO.NEW_TODAY' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Sessions Today' : 'جلسات اليوم' }}", "{{ 'CEO.SESSIONS_TODAY' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'attendance' : 'حضور' }}", "{{ 'CEO.ATTENDANCE' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Pending Payments' : 'المدفوعات المعلقة' }}", "{{ 'CEO.PENDING_PAYMENTS' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Insurance delays' : 'تأخير التأمين' }}", "{{ 'CEO.INSURANCE_DELAYS' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Revenue Trend (30 Days)' : 'اتجاه الإيرادات (30 يوماً)' }}", "{{ 'CEO.REVENUE_TREND' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Recent Activity' : 'النشاط الأخير' }}", "{{ 'CEO.RECENT_ACTIVITY' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'booked a session' : 'حجزت جلسة' }}", "{{ 'CEO.BOOKED_SESSION' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? '2 mins ago' : 'منذ دقيقتين' }}", "{{ 'CEO.MINS_AGO_2' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'completed assessment' : 'أكملت التقييم' }}", "{{ 'CEO.COMPLETED_ASSESSMENT' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? '15 mins ago' : 'منذ 15 دقيقة' }}", "{{ 'CEO.MINS_AGO_15' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'marked absent' : 'سُجل غائباً' }}", "{{ 'CEO.MARKED_ABSENT' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? '1 hour ago' : 'منذ ساعة' }}", "{{ 'CEO.HOUR_AGO_1' | translate }}"],

  // Receptionist
  ["{{ langService.currentLang() === 'en' ? 'Receptionist Dashboard' : 'لوحة تحكم موظف الاستقبال' }}", "{{ 'RECEPTIONIST.DASHBOARD_TITLE' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? \"Overview of today's clinic activity\" : 'نظرة عامة على نشاط العيادة اليوم' }}", "{{ 'RECEPTIONIST.DASHBOARD_SUB' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? \"Today's Sessions\" : 'جلسات اليوم' }}", "{{ 'RECEPTIONIST.TODAYS_SESSIONS' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'completed' : 'مكتملة' }}", "{{ 'RECEPTIONIST.COMPLETED_LC' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Present Doctors' : 'الأطباء الحاضرين' }}", "{{ 'RECEPTIONIST.PRESENT_DOCTORS' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'absent today' : 'غائبين اليوم' }}", "{{ 'RECEPTIONIST.ABSENT_TODAY' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Walk-ins' : 'زيارات بدون موعد' }}", "{{ 'RECEPTIONIST.WALKINS' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'High' : 'مرتفع' }}", "{{ 'RECEPTIONIST.HIGH' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'traffic' : 'ازدحام' }}", "{{ 'RECEPTIONIST.TRAFFIC' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Pending Payments' : 'مدفوعات معلقة' }}", "{{ 'RECEPTIONIST.PENDING_PAYMENTS' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Requires attention' : 'تتطلب الانتباه' }}", "{{ 'RECEPTIONIST.REQUIRES_ATTENTION' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? \"Today's Sessions Timeline\" : 'الجدول الزمني لجلسات اليوم' }}", "{{ 'RECEPTIONIST.TIMELINE_TITLE' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Time' : 'الوقت' }}", "{{ 'COMMON.TIME' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Patient' : 'المريض' }}", "{{ 'COMMON.PATIENT' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Doctor' : 'الطبيب' }}", "{{ 'COMMON.DOCTOR' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Room' : 'الغرفة' }}", "{{ 'COMMON.ROOM' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Status' : 'الحالة' }}", "{{ 'COMMON.STATUS' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Action' : 'إجراء' }}", "{{ 'COMMON.ACTION' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Completed' : 'مكتمل' }}", "{{ 'COMMON.COMPLETED' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'View' : 'عرض' }}", "{{ 'COMMON.VIEW' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'In Progress' : 'قيد التقدم' }}", "{{ 'COMMON.IN_PROGRESS' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Waiting' : 'انتظار' }}", "{{ 'COMMON.WAITING' | translate }}"],

  // Doctor
  ["{{ langService.currentLang() === 'en' ? \"Today's Schedule\" : 'جدول اليوم' }}", "{{ 'DOCTOR.DASHBOARD_TITLE' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Your upcoming sessions' : 'جلساتك القادمة' }}", "{{ 'DOCTOR.DASHBOARD_SUB' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Room 1' : 'غرفة 1' }}", "{{ 'DOCTOR.ROOM_1' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Room 3' : 'غرفة 3' }}", "{{ 'DOCTOR.ROOM_3' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Session 4 of 10' : 'جلسة 4 من 10' }}", "{{ 'DOCTOR.SESSION_4_10' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Session 8 of 12' : 'جلسة 8 من 12' }}", "{{ 'DOCTOR.SESSION_8_12' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Start Session' : 'بدء الجلسة' }}", "{{ 'DOCTOR.START_SESSION' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Available Slot' : 'موعد متاح' }}", "{{ 'DOCTOR.AVAILABLE_SLOT' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Assessment' : 'تقييم' }}", "{{ 'DOCTOR.ASSESSMENT' | translate }}"],

  // Senior
  ["{{ langService.currentLang() === 'en' ? 'Senior Dashboard' : 'لوحة تحكم كبير المعالجين' }}", "{{ 'SENIOR.DASHBOARD_TITLE' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Overview of assessments and clinic floor' : 'نظرة عامة على التقييمات وقسم العيادة' }}", "{{ 'SENIOR.DASHBOARD_SUB' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? \"Today's Tasks\" : 'مهام اليوم' }}", "{{ 'SENIOR.TODAYS_TASKS' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Assessments (4)' : 'تقييمات (4)' }}", "{{ 'SENIOR.ASSESSMENTS_4' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Re-assessments (2)' : 'إعادة التقييم (2)' }}", "{{ 'SENIOR.REASSESSMENTS_2' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Start' : 'بدء' }}", "{{ 'COMMON.START' | translate }}"],
  ["{{ langService.currentLang() === 'en' ? 'Live View - Doctors Capacity' : 'عرض مباشر - طاقة الأطباء' }}", "{{ 'SENIOR.LIVE_VIEW_CAPACITY' | translate }}"]
];

function replaceInFile(filePath) {
  let content = fs.readFileSync(filePath, 'utf8');
  for (const [find, replace] of mappings) {
    content = content.split(find).join(replace);
  }
  fs.writeFileSync(filePath, content, 'utf8');
}

const files = [
  'src/app/features/ceo/dashboard/dashboard.html',
  'src/app/features/receptionist/dashboard/dashboard.html',
  'src/app/features/doctor/dashboard/dashboard.html',
  'src/app/features/senior/dashboard/dashboard.html'
];

for (const file of files) {
  const f = path.join(__dirname, file);
  if (fs.existsSync(f)) {
    replaceInFile(f);
  } else {
    console.log('Not found:', f);
  }
}

File: replace_translations.py
import os
import re

files_to_process = [
    r'src\app\features\ceo\dashboard\dashboard.html',
    r'src\app\features\receptionist\dashboard\dashboard.html',
    r'src\app\features\doctor\dashboard\dashboard.html',
    r'src\app\features\senior\dashboard\dashboard.html',
    r'src\app\layout\sidebar\sidebar.html',
    r'src\app\layout\topbar\topbar.html',
    r'src\app\shared\components\language-selector\language-selector.ts'
]

# A dictionary to map the English string to its Translation Key
mapping = {
    'Executive Overview': "CEO.DASHBOARD_TITLE",
    'Real-time performance metrics': "CEO.DASHBOARD_SUB",
    'Export Report': "CEO.EXPORT_REPORT",
    "Today\\'s Revenue": "CEO.TODAYS_REVENUE",
    "Today's Revenue": "CEO.TODAYS_REVENUE",
    'vs yesterday': "CEO.VS_YESTERDAY",
    'Active Patients': "CEO.ACTIVE_PATIENTS",
    'new today': "CEO.NEW_TODAY",
    'Sessions Today': "CEO.SESSIONS_TODAY",
    'attendance': "CEO.ATTENDANCE",
    'Pending Payments': "CEO.PENDING_PAYMENTS",
    'Insurance delays': "CEO.INSURANCE_DELAYS",
    'Revenue Trend (30 Days)': "CEO.REVENUE_TREND",
    'Recent Activity': "CEO.RECENT_ACTIVITY",
    'booked a session': "CEO.BOOKED_SESSION",
    '2 mins ago': "CEO.MINS_AGO_2",
    'completed assessment': "CEO.COMPLETED_ASSESSMENT",
    '15 mins ago': "CEO.MINS_AGO_15",
    'marked absent': "CEO.MARKED_ABSENT",
    '1 hour ago': "CEO.HOUR_AGO_1",
    
    'Receptionist Dashboard': "RECEPTIONIST.DASHBOARD_TITLE",
    "Overview of today\\'s clinic activity": "RECEPTIONIST.DASHBOARD_SUB",
    "Overview of today's clinic activity": "RECEPTIONIST.DASHBOARD_SUB",
    "Today\\'s Sessions": "RECEPTIONIST.TODAYS_SESSIONS",
    "Today's Sessions": "RECEPTIONIST.TODAYS_SESSIONS",
    'completed': "RECEPTIONIST.COMPLETED_LC",
    'Present Doctors': "RECEPTIONIST.PRESENT_DOCTORS",
    'absent today': "RECEPTIONIST.ABSENT_TODAY",
    'Walk-ins': "RECEPTIONIST.WALKINS",
    'High': "RECEPTIONIST.HIGH",
    'traffic': "RECEPTIONIST.TRAFFIC",
    'Requires attention': "RECEPTIONIST.REQUIRES_ATTENTION",
    "Today\\'s Sessions Timeline": "RECEPTIONIST.TIMELINE_TITLE",
    "Today's Sessions Timeline": "RECEPTIONIST.TIMELINE_TITLE",
    
    'Time': "COMMON.TIME",
    'Patient': "COMMON.PATIENT",
    'Doctor': "COMMON.DOCTOR",
    'Room': "COMMON.ROOM",
    'Status': "COMMON.STATUS",
    'Action': "COMMON.ACTION",
    'Completed': "COMMON.COMPLETED",
    'View': "COMMON.VIEW",
    'In Progress': "COMMON.IN_PROGRESS",
    'Waiting': "COMMON.WAITING",
    'Start': "COMMON.START",
    
    "Today\\'s Schedule": "DOCTOR.DASHBOARD_TITLE",
    "Today's Schedule": "DOCTOR.DASHBOARD_TITLE",
    'Your upcoming sessions': "DOCTOR.DASHBOARD_SUB",
    'Room 1': "DOCTOR.ROOM_1",
    'Room 3': "DOCTOR.ROOM_3",
    'Session 4 of 10': "DOCTOR.SESSION_4_10",
    'Session 8 of 12': "DOCTOR.SESSION_8_12",
    'Start Session': "DOCTOR.START_SESSION",
    'Available Slot': "DOCTOR.AVAILABLE_SLOT",
    'Assessment': "DOCTOR.ASSESSMENT",
    
    'Senior Dashboard': "SENIOR.DASHBOARD_TITLE",
    'Overview of assessments and clinic floor': "SENIOR.DASHBOARD_SUB",
    "Today\\'s Tasks": "SENIOR.TODAYS_TASKS",
    "Today's Tasks": "SENIOR.TODAYS_TASKS",
    'Assessments (4)': "SENIOR.ASSESSMENTS_4",
    'Re-assessments (2)': "SENIOR.REASSESSMENTS_2",
    'Live View - Doctors Capacity': "SENIOR.LIVE_VIEW_CAPACITY",
    
    'English': "COMMON.ENGLISH"
}

pattern = re.compile(r'\{\{\s*langService\.currentLang\(\)\s*===\s*\'en\'\s*\?\s*([\'"])(.*?)\1\s*:\s*([\'"])(.*?)\3\s*\}\}')

def replacer(match):
    english_text = match.group(2)
    key = mapping.get(english_text)
    if not key:
        print(f"Warning: No mapping found for '{english_text}'")
        return match.group(0)
    return f"{{{{ '{key}' | translate }}}}"

for file_path in files_to_process:
    full_path = os.path.join(r"d:\TPC\admin panel\physical-therapy-project-Admin-Panel", file_path)
    if os.path.exists(full_path):
        with open(full_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        new_content = pattern.sub(replacer, content)
        
        if content != new_content:
            with open(full_path, 'w', encoding='utf-8') as f:
                f.write(new_content)
            print(f"Updated {file_path}")
    else:
        print(f"File not found: {full_path}")
